backend/src/error.rs

```diff
@@ -0,0 +1,225 @@
+use std::error::Error;
+
+use crate::data_entry::DataError;
+use axum::extract::rejection::JsonRejection;
+use axum::http::StatusCode;
+use axum::response::{IntoResponse, Response};
+use axum::Json;
+use hyper::header::InvalidHeaderValue;
+use serde::{Deserialize, Serialize};
+use sqlx::Error::RowNotFound;
+use tracing::error;
+use typst::diag::SourceDiagnostic;
+use utoipa::ToSchema;
+
+#[derive(Serialize, Deserialize, ToSchema, Debug)]
+pub enum ErrorReference {
+    EntryNumberNotSupported,
+    EntryNotFound,
+    PollingStationAlreadyFinalized,
+    PollingStationDataValidation,
+    InvalidVoteGroup,
+    InvalidVoteCandidate,
+    InvalidData,
+    EntryNotUnique,
+    Database,
+    Internal,
+    Pdf,
```

Comment:
These three feel a bit odd in this list, they do not describe what is wrong like the others. Maybe change them into for instance `DatabaseError`, `InternalServerError` and `PdfGenerationError`, to also have more clear names in the OpenAPI ?

Reply:
My reasoning was that the term Error is already in the enum name, but in this case it would make sense to be a bit more descriptive in the values I guess.